torch_spyre/csrc/spyre_mem.cpp

```diff
@@ -210,17 +212,16 @@ auto get_dim_device_size(int stick_size, int dim,
  * @param host2device: direction of data conversion
  * @return description of data conversion
  */
-auto get_device_stride_info(c10::IntArrayRef sizes, c10::IntArrayRef strides,
-                            int stick_size, bool host2device)
-    -> data_conversion_stride_info {
-  data_conversion_stride_info stride_info;
-  auto cpu_shape = sizes.vec();
-  auto cpu_strides = strides.vec();
-  auto dev_dim_order = get_device_layout(sizes);
-  auto requires_padding = (cpu_shape[dev_dim_order.front()] % stick_size != 0);
+auto get_device_stride_info(std::vector<int64_t> cpu_shape,
+                            std::vector<int64_t> cpu_strides,
+                            std::vector<int64_t> dev_shape, int stick_size,
```

Comment:
why are we passing the shapes and strides by copy? They're not big vectors by any means, but still